source/agora/node/Ledger.d

```diff
@@ -305,15 +306,14 @@ public class Ledger
     private void addValidatedBlock (const ref Block block) @safe
     {
         if (!this.storage.saveBlock(block))
-            assert(0);
+            assert(0, format!"Failed to save block: %s"(prettify(block)));
 
         ManagedDatabase.beginBatch();
         scope (failure) ManagedDatabase.rollback();
-
-        auto old_count = this.enroll_man.validatorCount();
         this.updateUTXOSet(block);
         this.updateValidatorSet(block);
         ManagedDatabase.commitBatch();
+        auto old_count = this.enroll_man.validatorCount();
```

Comment:
That's a bug. `old_count` **needs** to be before `updateValidatorSet`. Check the `validators_changed` code to see why.

Reply:
Good catch. Correct in #1273